app/components/mapbox-gl-draw.js

```diff
@@ -32,8 +32,10 @@ export const DefaultDraw = MapboxDraw.bind(null, {
     'draw_annotations:linear': AnnotationsMode, // These are identical because they function the same
     'draw_annotations:curved': AnnotationsMode, // but only really need to be named differently
     'draw_annotations:square': AnnotationsMode,
-    'draw_annotations:label': AnnotationsDrawPoint,
-    'draw_annotations:centerline': AnnotationsDrawPoint,
+    'draw_annotations:label': AnnotationsDrawPointMode,
+    'draw_annotations:centerline': AnnotationsDrawPointMode,
+    // duplicate mode with distinct name  to avoid `skipToDirectSelect` trigger when we switch to simple_select for delete
+    simple_select_delete: MapboxDraw.modes.simple_select,
```

Comment:
Just a general thought: are we abusing the custom modes feature here? It's really used as a way to determine _how_ something was produced without the mode itself having anything to do with the output it was given (for example, AnnotationsMode is the same for linear, curved, and square, which are all rendered later based on the "imprinted" mode you see in the hash).